### DISCUSSION TITLE
Is it safe to reuse jobId across queues?

### DISCUSSION THREAD

**Discussion by beinoriusju** [taskforcesh/bullmq#2659] [Q&A] [answered]
I've noticed that **jobId** needs to be unique per queue not globally. This helps me move jobs between queues and still be able to query them using the same jobId. Are there any gochas here that I don't know and do you agree it needs to be mentioned in documentation? (eg _"unique per que"_)

https://github.com/taskforcesh/bullmq/blob/master/docs/gitbook/guide/jobs/job-ids.md

**Comment by manast** [comment, marked as answer]
There are no gotchas, the jobs keys are prefixed by the queue name.
